src/helper/client.cpp

```diff
@@ -474,19 +475,39 @@ bool client::handle_command(network::request_shutdown::request &command)
 
 bool client::run_client_init()
 {
-    static constexpr auto client_init_timeout{std::chrono::milliseconds{500}};
-    return handle_message<network::client_init>(
-        *this, *broker_, client_init_timeout);
+    try {
+        static constexpr auto client_init_timeout{
+            std::chrono::milliseconds{500}};
+        return handle_message<network::client_init>(
+            *this, *broker_, client_init_timeout);
+    } catch (const unexpected_command &) {
+        send_error_response(*broker_);
+    }
+
+    return false;
 }
 
 bool client::run_request()
 {
-    // TODO: figure out how to handle errors which require sending an error
-    //       response to ensure the extension doesn't hang.
-    return handle_message<network::request_init, network::request_exec,
-        network::config_sync, network::request_shutdown>(
-        *this, *broker_, std::chrono::milliseconds{0} /* no initial timeout */
-    );
+    try {
+        if (!request_enabled_) {
+            return handle_message<network::request_init, network::config_sync>(
+                *this, *broker_, std::chrono::milliseconds{0}
+                /* no initial timeout */
+            );
+        }
+        // TODO: figure out how to handle errors which require sending an error
+        //       response to ensure the extension doesn't hang.
+        return handle_message<network::request_init, network::request_exec,
+            network::config_sync, network::request_shutdown>(*this, *broker_,
+            std::chrono::milliseconds{0} /* no initial timeout */
+        );
+    } catch (const unexpected_command &) {
+        compute_client_status();
```

Comment:
Unfortunately this isn't enough because the client could send messages resulting in other valid errors. For example `std::out_of_range` or `std::bad_cast`, I think it might be best to just compute the client status inside `handle_message`, I guess you can compute the client status only when result is true after the try catch (since that is the error handling path).

Reply:
ok